platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/transition/execution/auto/WorkflowAutoTransitionExecutor.java

```diff
@@ -27,44 +27,61 @@ public WorkflowAutoTransitionExecutor(AGWorkflowItem item, Collection<AGWorkflow
 	public void evaluateAndExecute() {
 
 		try (DbTransaction transaction = new DbTransaction()) {
-			checkSourceNodeOfTransitions();
-			List<AGWorkflowTransition> validTransitions = transitions//
-				.stream()
-				.filter(it -> validateNodePreconditions(item, it))
-				.collect(Collectors.toList());
-			if (validTransitions.size() > 1) {
-				List<IDisplayString> transitionTitles = validTransitions//
+			item.reloadForUpdate();
+			if (checkSourceNodeOfTransitions()) {
+				List<AGWorkflowTransition> validTransitions = transitions//
 					.stream()
-					.map(AGWorkflowTransition::toDisplayWithoutId)
+					.filter(it -> validateNodePreconditions(item, it))
 					.collect(Collectors.toList());
-				throw new SofticarUserException(//
-					WorkflowI18n.WORKFLOW_ITEM_ARG1_HAS_MORE_THAN_ONE_EXECUTABLE_AUTO_TRANSITION_ARG2
-						.toDisplay(item.toDisplayWithoutId(), Imploder.implode(transitionTitles, "\n")));
-			} else if (validTransitions.size() == 1) {
-				var transition = validTransitions.get(0);
-				var targetNode = transition.getTargetNode();
-				Log
-					.finfo(//
-						"Executing transition '%s' into target node '%s' for item '%s'.",
-						transition.toDisplayWithoutId(),
-						targetNode.toDisplayWithoutId(),
-						item.toDisplay());
-				transition.executeSideEffect(item);
-				new WorkflowTaskManager(item).setNextNodeAndGenerateTasks(targetNode);
-				new AGWorkflowAutoTransitionExecution().setWorkflowItem(item).setWorkflowTransition(transition).save();
-			} else {
-				Log.fverbose("No executable auto transition found.");
+				evaluateAndExecute(validTransitions);
 			}
 			transaction.commit();
 		}
 	}
 
-	private void checkSourceNodeOfTransitions() {
+	private void evaluateAndExecute(List<AGWorkflowTransition> validTransitions) {
+
+		if (validTransitions.size() > 1) {
+			List<IDisplayString> transitionTitles = validTransitions//
+				.stream()
+				.map(AGWorkflowTransition::toDisplayWithoutId)
+				.collect(Collectors.toList());
+			throw new SofticarUserException(//
+				WorkflowI18n.WORKFLOW_ITEM_ARG1_HAS_MORE_THAN_ONE_EXECUTABLE_AUTO_TRANSITION_ARG2
+					.toDisplay(item.toDisplayWithoutId(), Imploder.implode(transitionTitles, "\n")));
+		} else if (validTransitions.size() == 1) {
+			var transition = validTransitions.get(0);
+			var targetNode = transition.getTargetNode();
+			Log
+				.finfo(//
+					"Executing transition '%s' into target node '%s' for item '%s'.",
+					transition.toDisplayWithoutId(),
+					targetNode.toDisplayWithoutId(),
+					item.toDisplay());
+			transition.executeSideEffect(item);
+			new WorkflowTaskManager(item).setNextNodeAndGenerateTasks(targetNode);
+			new AGWorkflowAutoTransitionExecution().setWorkflowItem(item).setWorkflowTransition(transition).save();
+		} else {
+			Log.fverbose("No executable auto transition found.");
+		}
+	}
+
+	private boolean checkSourceNodeOfTransitions() {
```

Comment:
I think we should rename this method to `allTransitionsHaveExpectedSourceNode`.
The reference then reads like:
`if (allTransitionsHaveExpectedSourceNode())`
...which is nice.

Reply:
done